lib/spack/spack/fetch_strategy.py

```diff
@@ -318,6 +318,11 @@ def fetch(self):
         url = None
         errors = []
         for url in self.candidate_urls:
+            if url[0:5] == 'azure':
+                import spack.util.azure_blob as azure_blob_util
```

Comment:
This import might become an issue here, since an import is an implicit assignment of a variable. Here a simple example with the same structure:
```python
Python 3.8.10 (default, Nov 26 2021, 20:14:08) 
[GCC 9.3.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import os
>>> def foo():
...   print('Before')
...   dir(os)
...   import os.path
>>> foo()
Before
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "<stdin>", line 3, in foo
UnboundLocalError: local variable 'os' referenced before assignment
```

Reply:
I think the import spack.util.azure_blob as azure_blob_util will work, similar syntx is used throughput the fetch_strategy.py file.
(e.g import spack.util.s3 as s3_util)